docs/content/guides/developer/advanced/randomness-onchain.mdx

```diff
@@ -0,0 +1,140 @@
+
+:::
+
+## Instantiating `RandomGenerator`
+
+`RandomGenerator` is secure as long as it's created by the consuming module. If passed as an argument, the caller might be able to predict the outputs of that `RandomGenerator` instance (for example, by calling `bcs::to_bytes(&generator)` and parsing its internal state).
+
+:::note
+
+The Move compiler prevents defining `public` functions with `RandomGenerator` as an argument.
+
+:::
+
+## Limited resources and `Random` dependent flows
+
+Developers should be aware that some resources that are available to transactions are limited. If a function that reads `Random` consumes more resources in the unhappy path flow than in the happy path flow, an attacker can use that difference to revert the transaction in the unhappy flow as previously demonstrated. Concretely, gas is such a resource. Consider the following code:
+
+```rust
+entry fun play(r: &Random, payment: Coin<SUI>, ...) {
+	...
+  let generator = new_generator(r, ctx);
+	let win = random::generate_bool(generator);
+	if (win) { // happy flow
+		... cheap computation ...
+	} else {
+		... very expensive computation ...
+	}
+}
+```
+
+Observe that the gas costs of a transaction that calls `play` depends on the value of `win` - An attacker could call this function with a gas object that has enough balance to cover the happy flow but not the unhappy one, resulting in it either winning or reverting the transaction (but never losing the payment).
+
+In many cases this is not an issue, like when selecting a raffle winner, lottery numbers, or a random NFT. However, in the cases where it can be problematic, you can do one of the following:
+
+- Write the function in a way that the happy flow consumes more gas than the unhappy one.
+    - Keep in mind that external functions or native ones can change in the future, potentially resulting in different costs compared to the time you conducted your tests.
+    - Use [profile-transaction](../../../references/cli/client.mdx#profile-a-transaction) on Testnet transactions to verify the costs of different flows.
+- Split the logic in two: one function that fetches a random value and stores it in an object, and another function that reads that stored value and completes the operation. The latter function might indeed fail, but now the random value is fixed and cannot be modified using repeated calls.
+
+See [random_nft.move](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/nfts/sources/random_nft.move) for examples.
+
+Other limited resources per transaction are:
+
+- The number of new objects.
+- The number of objects that can be used (including dynamic fields).
+
+## Related Links
+
+- [Sui Framework Reference](../../../references/framework)
+	- <a href="/references/framework/sui-framework/random" data-noBrokenLinkCheck='true'>random.move</a>
+- [Randomness example](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/nfts/sources/random_nft.move)
```

Comment:
another one to add here (simpler)

Reply:
Indention? Intentional because random.move is a child of framework reference